chore: Bump network-controller to v27 (+ dependencies)

mcmire opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Upgrade @metamask/network-controller to v27 with related dependency updates and adjust TransactionController init to handle the new network state type.
> 
> - **Dependencies**:
>   - Bump `@metamask/network-controller` to `^27.0.0`.
>   - Align peers: `@metamask/controller-utils` → `^11.16.0`, `@metamask/eth-json-rpc-middleware` → `^22.0.0`, `@metamask/json-rpc-engine` → `^10.2.0`.
> - **Engine · TransactionController init (`app/core/Engine/controllers/transaction-controller/transaction-controller-init.ts`)**:
>   - Add compatibility handling/comments for `getNetworkState` due to `NetworkStatus.Degraded` added in `@metamask/network-controller@v27`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e08b54afdb36ac835e505bdc91cfac224512ed7c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

mcmire committed:
WIP - More optimized Metro settings

mcmire committed:
Bump network-controller to v26 (+ dependencies)

mcmire committed:
Cover this one too

Commit:
fix: increase Node.js heap size for iOS bundle generation in CI

The js-bundle-size-check job was experiencing JavaScript heap out of memory errors during the iOS bundle generation step. This adds NODE_OPTIONS with --max_old_space_size=8192 (8GB) to prevent these failures, following the same pattern used in other memory-intensive CI jobs.

mcmire committed:
Merge branch 'main' into bump-network-controller-to-v26

mcmire committed:
Merge branch 'main' into bump-network-controller-to-v26

mcmire committed:
Update preview builds

mcmire committed:
Revert "WIP - More optimized Metro settings"

This reverts commit 68799b786145f3c092ff8e0c7d328de762356ccb.

mcmire committed:
dedupe

mcmire committed:
Merge branch 'main' into bump-network-controller-to-v26

cryptodev-2s committed:
Merge remote-tracking branch 'origin/main' into bump-network-controller-to-v26

cryptodev-2s committed:
Remove preview builds

cryptodev-2s committed:
fix: Add yarn resolution for @metamask/network-controller

- Force all packages to use @metamask/network-controller ^27.0.0
- Resolves TypeScript error where transaction-controller's bundled network-controller was missing NetworkStatus.Degraded property
- Ensures consistent network-controller types across all MetaMask packages

mikesposito commented on package.json:
```diff
@@ -177,7 +177,8 @@
     "react-native@0.76.9": "patch:react-native@npm%3A0.76.9#./.yarn/patches/react-native-npm-0.76.9-1c25352097.patch",
-    "@metamask/transaction-controller@npm:^62.3.1": "patch:@metamask/transaction-controller@npm%3A62.3.1#~/.yarn/patches/@metamask-transaction-controller-npm-61.0.0-cccac388c7.patch"
+    "@metamask/transaction-controller@npm:^62.3.1": "patch:@metamask/transaction-controller@npm%3A62.3.1#~/.yarn/patches/@metamask-transaction-controller-npm-61.0.0-cccac388c7.patch",
+    "@metamask/network-controller": "^27.0.0"
```
Why are we forcing the resolution here?

mikesposito replied:
I think there is an incompatibility with the type expected by transaction-controller due to the network-controller version it depends upon. We usually use `@ts-expect-error` for this kind of type issues - assuming we are sure that this incompatibility doesn't break functionality

cryptodev-2s replied:
Force resolution removed here https://github.com/MetaMask/metamask-mobile/pull/22911/commits/cd0471327069a208575ba81d8c76c8998911e0f9

cryptodev-2s committed:
chore: remove network-controller force resolution

Remove the force resolution for @metamask/network-controller and add
ts-expect-error comment to handle type mismatch with NetworkStatus enum.

The latest version (v27.0.0+) adds NetworkStatus.Degraded enum value
which causes type mismatches in our codebase.

See: https://github.com/MetaMask/core/pull/7186

cryptodev-2s committed:
Merge branch 'main' into bump-network-controller-to-v26
